### ISSUE TITLE
Terraform doesn't support event subscription in DMS

### ISSUE THREAD

**Issue by Rahimovs** [terraform-providers/terraform-provider-aws#3057]
Please add ability to add DMS event subscription (https://us-west-2.console.aws.amazon.com/dms/home?region=us-west-2#event-subscriptions:) 

https://docs.aws.amazon.com/cli/latest/reference/dms/create-event-subscription.html
Terraform documentation checked: 
https://www.terraform.io/docs/providers/aws/r/dms_replication_task.html

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

Terraform v0.10.2

### Affected Resource(s)
Please list the resources as a list, for example:
- opc_instance
- opc_storage_volume

Database Migration Service

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```
n/a

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

n/a

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

n/a

### Expected Behavior
What should have happened?

n/a

### Actual Behavior
What actually happened?

n/a

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

n/a

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

n/a

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

n/a

**Comment by vide** [comment]
@mwarkentin thanks for the CF trick, it worked for us too! Really hope this ticket gets some traction in the future

**Comment by bflad** [comment]
Support for a new `aws_dms_event_subscription` resource has been merged and will release with version 2.58.0 of the Terraform AWS Provider, later this week. Thanks to @hawknewton for the implementation. 👍
